Provide cabal bounds in compliance with hackage requirements

abailly opened the pull request:
This small PR is intended as preparatory step to upload cardano-crypto to hackage.haskell.org, which is itself a pre-requisite to upload cardano-addresses to Hackage.
See https://hackage.haskell.org/package/cardano-crypto-1.2.0/candidate for how it would look like on Hackage.

#99

abailly committed:
Provide cabal bounds in compliance with hackage requirements

This small PR is intended as preparatory step to upload cardano-crypto
to hackage.haskell.org, which is itself a pre-requisite to upload
cardano-addresses to Hackage.

erikd reviewed (approved):
LGTM!

abailly commented:
I wonder if those bounds are not too tight 🤔

erikd commented:
> I wonder if those bounds are not too tight 🤔

Looks fine to me. Try running `cabal outdated` on it. If that does not complain, anyone who thinks they are too tight can fix that in the `cabal.project` file with an `allow-newer` or `allow-older`. They can then submit a PR.

abailly committed:
Relax bounds

It seems the bounds set by cabal gen-bounds are rather tight, and that
quite a few dependencies have not any update for years so there's no
need to be too strict in the bounds.

abailly commented:
Thanks @erikd. As the dependency tree is not large, I had a quick look at all of them to get a sense of what would be sensible more relaxed bounds. Here is what the dependency looks like

![deps](https://github.com/user-attachments/assets/887b84a2-e3f8-4af3-aaf9-b0a1b6bbf69f)

abailly commented:
Anything preventing us to merge this? @erikd ? We would like to publish cardano-addresses to hackage which requires publishing _this_ to Hackage.

abailly commented:
@disassembler @Ranchhand87 We need this to be able to publish cardano-address to hackage
